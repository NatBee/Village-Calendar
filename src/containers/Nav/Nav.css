.nav {
  font-family: 'Playfair Display', serif;
  background: #89988a;
  color: #DBE3D4;
  justify-content: center;
  position: fixed;
  top: 0;
  width: 100%;
  overflow: hidden;
  z-index: 1;
  display: flex;
  justify-content: space-between; }

.nav h1 {
  margin: 0;
  padding: 35px 0px 8px 25px; }

.nav button {
  background-color: #3E4D4E;
  height: 30px;
  color: #DBE3D4;
  font-family: 'Exo', sans-serif;
  border: 2px solid #293435;
  width: 100px; }

.nav button:hover {
  background-color: #293435; }

.display-name {
  padding: 10px 20px 5px; }

.login {
  margin: 40px 20px 0; }

.logout-btn {
  margin: 5px 0 15px 90px; }
