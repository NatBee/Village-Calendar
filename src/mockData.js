export const mockData = {
  user: {
    user: 'Dk', 
    credential: 'Xg', 
    additionalUserInfo: 'Pg', 
    operationType: "signIn"
  },
  events: [
    {
      start: "2018-02-09T09:00:00-07:00",
      end: "2018-02-09T10:00:00-07:00",
      title: "Super Fun Party",
      description: "Fun with friends",
      eventID: "7js6e24fo0kin4qcotoegu37hg",
      location: 'Mi Casa'
    }
  ],
  token: 'token_number',
  calendarID: 'calendarID',
  village: [{natalie: 'nataliesmontoya@gmail.com'}],
  time: {
    startTime: "2018-02-28T19:00:00.000Z", 
    endTime: "2018-02-28T20:00:00.000Z"
  },
  event: {
    start: "2018-02-09T09:00:00-07:00",
    end: "2018-02-09T10:00:00-07:00",
    title: "Super Fun Party",
    description: "Fun with friends",
    eventID: "7js6e24fo0kin4qcotoegu37hg",
    location: 'Mi Casa'
  }
}