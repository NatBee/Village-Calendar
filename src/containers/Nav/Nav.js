import React, { Component } from 'react';
import { logOutUser, removeAllEvents, removeToken, logInUser, setToken, removeCalendarID, removeVillageList } from '../../actions/index';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import './Nav.css';
import firebase from 'firebase/app';
import 'firebase/auth';
import { config } from '../../helper/apiKey';
import PropTypes from 'prop-types';

firebase.initializeApp(config);

class Nav extends Component {

  componentDidMount = () => {
    const token = JSON.parse(localStorage.getItem('ouath2-access-token'));
    if(token) {
      this.props.setToken(token)
    }
  }

  logIn = async () => {
    const provider = new firebase.auth.GoogleAuthProvider();
    provider.addScope('https://www.googleapis.com/auth/calendar');
    const auth = firebase.auth();
    const authentication = await auth.signInWithPopup(provider);
    this.props.logInUser(authentication);
    const token = authentication.credential.accessToken;
    localStorage.setItem('ouath2-access-token', JSON.stringify(token));
    this.props.setToken(token);
    this.pageDirect();
  } 

  pageDirect = () => {
    if(this.props.history.location.pathname === '/') {
      this.props.history.push('/register');
    } else if(this.props.history.location.pathname === '/calendar') {
      this.props.history.push('/calendar')
    } else {
      this.props.history.push('/register');
    }
  } 

  buttonDisplay = () => {
    if(this.props.token === '') {
      return(
        <button className="login" onClick={this.logIn}>Google Log In</button>
      )
    } else {
      return(
        <div>
          <button onClick={this.logOut}>Log Out</button>
        </div>
      )
    }
  }

  logOut = () => {
    firebase.auth().signOut();
    const user = this.props.user;
    this.props.logOutUser(user);
    this.props.removeAllEvents(this.props.events);
    this.props.removeToken(this.props.token);
    this.props.removeCalendarID(this.props.calendarID);
    this.props.removeVillageList(this.props.village);
    localStorage.clear();
    this.props.history.push('/');
  }

  render() {
    return(
      <div>
        <h1>VILLAGE</h1>
        {this.buttonDisplay()}
      </div>
    )
  }
}

export const mapStateToProps = (store) => ({
  user: store.user,
  events: store.events,
  token: store.token,
  calendarID: store.calendarID,
  village: store.village
})

export const mapDispatchToProps = (dispatch) => ({
  logOutUser: (user) => dispatch(logOutUser(user)),
  removeAllEvents: (events) => dispatch(removeAllEvents(events)),
  removeToken: (token) => dispatch(removeToken(token)),
  logInUser: (user) => dispatch(logInUser(user)),
  setToken: (token) => dispatch(setToken(token)),
  removeCalendarID: (calendarID) => dispatch(removeCalendarID(calendarID)),
  removeVillageList: (village) => dispatch(removeVillageList(village))
})

Nav.propTypes = {
  user: PropTypes.object,
  events: PropTypes.array,
  token: PropTypes.string,
  calendarID: PropTypes.string,
  village: PropTypes.array,
  logOutUser: PropTypes.func,
  removeAllEvents: PropTypes.func,
  removeToken: PropTypes.func,
  logInUser: PropTypes.func,
  setToken: PropTypes.func,
  removeCalendarID: PropTypes.func,
  removeVillageList: PropTypes.func,
  login: PropTypes.func,
  pageDirect: PropTypes.func,
  buttonDisplay: PropTypes.func,
  logOut: PropTypes.func,
  history: PropTypes.object,
  location: PropTypes.object.isRequired,
}  


export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Nav));