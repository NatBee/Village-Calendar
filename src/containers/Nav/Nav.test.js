/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import { Nav, mapStateToProps, mapDispatchToProps } from './Nav';
import { mockData } from '../../mockData';

global.localStorage = {
  getItem(keyword) {
    if (!global.localStorage[keyword]) {
      return null;
    }
    return JSON.stringify(global.localStorage[keyword]);
  },
  setItem(keyword, value) {
    global.localStorage[keyword] = value;
  }
};

describe('Nav', () => {

  it('should match the snapshot', () => {
    localStorage.setItem('ouath2-access-token', mockData.token);
    const wrapper = shallow(<Nav setToken={jest.fn()}/>);
    expect(wrapper).toMatchSnapshot();
  })

  it('should map to the store correctly', () => {
    const mockStore = {
      user: mockData.user,
      events: mockData.events,
      token: mockData.token,
      calendarID: mockData.calendarID,
      village: mockData.village
    }
    const mapped = mapStateToProps(mockStore);
    expect(mapped).toEqual(mockStore);
  })

  it('should call the dispatch function when using logOutUser function from mapDispatchToProps', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.logOutUser();
    expect(mockDispatch).toHaveBeenCalled();
  })

  it('should call the dispatch function when using removeAllEvents function from mapDispatchToProps', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.removeAllEvents();
    expect(mockDispatch).toHaveBeenCalled();
  })

  it('should call the dispatch function when using removeToken function from mapDispatchToProps', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.removeToken();
    expect(mockDispatch).toHaveBeenCalled();
  })

  it('should call the dispatch function when using logInUser function from mapDispatchToProps', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.logInUser();
    expect(mockDispatch).toHaveBeenCalled();
  })

  it('should call the dispatch function when using setToken function from mapDispatchToProps', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.setToken();
    expect(mockDispatch).toHaveBeenCalled();
  })

  it('should call the dispatch function when using removeCalendarID function from mapDispatchToProps', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.removeCalendarID();
    expect(mockDispatch).toHaveBeenCalled();
  })

  it('should call the dispatch function when using removeVillageList function from mapDispatchToProps', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.removeVillageList();
    expect(mockDispatch).toHaveBeenCalled();
  })
})