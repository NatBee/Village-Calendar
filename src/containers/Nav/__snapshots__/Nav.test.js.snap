// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    setToken={
      [MockFunction] {
        "calls": Array [
          Array [
            "ya29.GlxpBSxcPM6wYT3djOnQ7SjuuiI9r3CZZlhqq0njhl3mVFNi59h3tJ20qwmOnpf5wNdRh1aBweKC0s3N5VPxYRioKTGir3EB5-Znpn0U7yyACYskQIqc9NAZbX2QhA",
          ],
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Village Calendar
        </h1>,
        <div>
          <button
            onClick={[Function]}
          >
            Log Out
          </button>
        </div>,
      ],
      "className": "nav",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Village Calendar",
        },
        "ref": null,
        "rendered": "Village Calendar",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            onClick={[Function]}
          >
            Log Out
          </button>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Log Out",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Log Out",
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Village Calendar
          </h1>,
          <div>
            <button
              onClick={[Function]}
            >
              Log Out
            </button>
          </div>,
        ],
        "className": "nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Village Calendar",
          },
          "ref": null,
          "rendered": "Village Calendar",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              onClick={[Function]}
            >
              Log Out
            </button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Log Out",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Log Out",
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
